frontend/src/lib/logic/inAppPrompt/inAppPromptLogic.tsx

```diff
@@ -0,0 +1,454 @@
+                        valid.push({ sequence, state: { step: 0 } })
+                    }
+                }
+            }
+            actions.setValidSequences(valid)
+        },
+        setValidSequences: () => actions.runFirstValidSequence({}),
+        runFirstValidSequence: ({ options }) => {
+            if (values.validSequences) {
+                actions.closePrompts()
+                let firstValid = null
+                if (options.runDismissedOrCompleted) {
+                    firstValid = values.validSequences[0]
+                } else {
+                    firstValid = values.validSequences.filter(
+                        (sequence) => !sequence.state.completed && !sequence.state.dismissed
+                    )?.[0]
+                }
+                if (firstValid) {
+                    const { sequence, state } = firstValid
+                    setTimeout(
+                        () => actions.runSequence(sequence, options.restart ? 0 : state.step),
+                        NEW_SEQUENCE_DELAY
+                    )
+                }
+            }
+        },
+        dismissSequence: () => {
+            if (values.sequenceKey) {
+                const key = values.sequenceKey
+                const currentState = values.userState[key]
+                if (currentState && !currentState.completed) {
+                    actions.updatePromptState({
+                        dismissed: true,
+                    })
+                    if (values.currentStep < values.prompts.length) {
+                        inAppPromptEventCaptureLogic.actions.reportPromptSequenceDismissed(
+                            values.sequenceKey,
+                            values.currentStep,
+                            values.prompts.length
+                        )
+                    }
+                }
+                actions.clearSequence()
+            }
+        },
+        setUserState: () => actions.syncState({}),
+        promptAction: ({ action }) => {
+            switch (action) {
+                case 'skip':
+                    actions.closePrompts()
+                    actions.skipTutorial()
+                    inAppPromptEventCaptureLogic.actions.reportTutorialSkipped()
+                    break
+                default:
+                    break
+            }
+        },
+        //@ts-expect-error
+        [router.actions.locationChanged]: () => {
+            actions.closePrompts()
+            actions.findValidSequences()
+        },
+    })),
+    events(({ actions, cache }) => ({
+        afterMount: () => {
+            posthog.onFeatureFlags(async (_, variants) => {
+                if (variants[FEATURE_FLAGS.IN_APP_PROMPTS_EXPERIMENT] === 'test') {
+                    actions.syncState({ forceRun: true })
+                }
+            })
+        },
+        beforeUnmount: [
+            () => {
+                cache.runOnClose?.()
+            },
+        ],
+    })),
```

Comment:
I believe these have their own top-level builders so you could write it:

```suggestion
    afterMount(({ actions }) => {
        posthog.onFeatureFlags(async (_, variants) => {
            if (variants[FEATURE_FLAGS.IN_APP_PROMPTS_EXPERIMENT] === 'test') {
                actions.syncState({ forceRun: true })
            }
        })
    }),
    beforeUnmount(({ cache }) => cache.runOnClose?.()),
```